File: Liskovs/violations/UserService.java
public class UserService {
    public void setAge(int age) {
        if (age < 0) {
            throw new IllegalArgumentException("Age must be non-negative");
        }
        System.out.println("Age set to " + age);
    }
    public static void main(String[] args) {
        UserService service = new AdultUserService();
        service.setAge(10);
    }

}


class AdultUserService extends UserService {
    @Override
    public void setAge(int age) {
        if (age < 18) {
            throw new IllegalArgumentException("User must be an adult");
        }
        System.out.println("Adult age set to " + age);
    }
}

File: Liskovs/violations/UserService.cs
using System;

public class UserService
{
    public virtual void SetAge(int age)
    {
        if (age < 0)
        {
            throw new ArgumentException("Age must be non-negative");
        }

        Console.WriteLine("Age set to " + age);
    }

    public static void Main(string[] args)
    {
        UserService service = new AdultUserService();
        service.SetAge(10); // Runtime exception
    }
}

public class AdultUserService : UserService
{
    public override void SetAge(int age)
    {
        if (age < 18)
        {
            throw new ArgumentException("User must be an adult");
        }

        Console.WriteLine("Adult age set to " + age);
    }
}
